Source/ISHRemote/Trisoft.ISHRemote/Cmdlets/Folder/GetIshFolder.cs

```diff
@@ -326,19 +356,16 @@ private void RetrieveRecursive(IshFolder ishFolder, int currentDepth, int maxDep
         {
             // put them on the pipeline depth-first-traversel
             string folderName = ishFolder.IshFields.GetFieldValue("FNAME", Enumerations.Level.None, Enumerations.ValueType.Value);
-            WriteVerbose(new string('>', currentDepth) + IshSession.FolderPathSeparator + folderName + IshSession.FolderPathSeparator);
 
             // only return IshFolder objects to the pipeline if the filter is either not set, or the current filter passes the filter criteria
-            if (FolderTypeFilter != null && FolderTypeFilter.Length > 0)
+            if ((FolderTypeFilter == null) || ((FolderTypeFilter != null) && FolderTypeFilter.Contains<Enumerations.IshFolderType>(ishFolder.IshFolderType)))
```

Comment:
FolderTypeFilter equals null OR FolderTypeFilter not-equals null... is always true.... and as the matching `else` no longer holds a WriteObject I wonder if the regular recurse still works without FolderTypeFilter

Reply:
Or the statement is too complex for my simple brain, and it still accurate :-)

Reply:
This implementation will write object in 2 cases OR-OR:
1) FolderTypeFilter is not provided (==null)
2) FolderTypeFilter is provided AND folder matches filter
This change has tests so should work.. :) 
But on the other hand complex If statements are not good either(as you had questions :)
 I cat try to rewrite to something that you had with nested If(s) and with logging about skipped folders

Reply:
..decomposed If statement into simpler if statements